### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by sivayogasubramanian**
Part of https://github.com/AY2122S1-CS2103T-W16-2/tp/issues/206.

[Preview here](https://sivayogasubramanian.github.io/tp/DeveloperGuide.html)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `edit-o 1`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `edit-o abc`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `edit-o 1 op/-2`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+
+1. Finding orders by keywords
+    1. Prerequisites: There are exactly 2 orders in the order list.
+       1. Suppose the first order has the following details:
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1472]:
```suggestion
       1. First order has the following details:
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Client Name: Bernice Yu
+          * Client Phone: 98762159
+          * Client Address: Blk 12 Holland Village Street 1. #01-40
+          * Recipe Name: Nasi Lemak
+          * Recipe Ingredients: Rice x 200 g, Fish x 1 whole
+          * Order Price: 3
+          * Order Deadline: 15 December 2021 at 1 pm
+          * Order Quantity: 2
+          * Completion status: Uncompleted
+       1. Suppose the second order has the following details:
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1482]:
```suggestion
       1. Second order has the following details:
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       1. Order list does not contain an order with the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
+          * Order Deadline: 12 December 2021 at 12 pm
+          * Order Quantity: 1
+       1. Client bookmark list contains at least one client and suppose the first client in the list has the following details:
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1333]:
```suggestion
       1. Client bookmark list contains at least one client and the first client in the list has the following details:
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Client Email: alexyeoh@gmail.com
+       1. Recipe bookmark list contains at least one recipe and suppose the first recipe in the list has the following details:
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1338]:
```suggestion
       1. Recipe bookmark list contains at least one recipe and the first recipe in the list has the following details:
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       <br> Expected: No order is added. Error details shown in the result display box.
+    1. Test case: `add-o c/0 r/1 op/4 od/12-12-2021 1200 oq/1`
+       <br> Expected: No order is added. Error details shown in the result display box.
+    1. Test case: `add-o op/4 od/12-12-2021 1200 oq/1`
+       <br> Expected: No order is added. Error details shown in the result display box.
+    1. Test case: `add-o`
+       <br> Expected: No order is added. Error details shown in the result display box.
+
+1. Adding an order ingredient
+    1. Prerequisites: Order list shows at least 1 order and at most 3 orders.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1372]:
```suggestion
    1. Prerequisites: Order list shows at least 1 order and at most 3 orders. The first order in the list does not have rice as one of its ingredients
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       <br> Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `add-oi 1 in/Chicken iq/0 iu/whole`
+       <br> Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `add-oi 1 in/Chicken iq/500000 iu/whole`
+       <br> Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `add-oi 1 in/Chi$ken iq/4 iu/whole`
+       <br> Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `add-oi 1 in/Chicken iq/4 iu/who%le`
+       <br> Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `add-oi in/Rice iq/400 iu/g 1`
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1386]: Remove this test case, it is the same as the one in line 1390

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Deleting an order
+    1. Prerequisites: Order list shows at least 1 order and at most 3 orders.
+    1. Test case: `delete-o 1`
+       <br>Expected: First order is deleted from the order list. Details of the deleted order are
+       shown in the result display box.
+    1. Test case: `delete-o -1`
+       <br>Expected: No order is deleted. Error details shown in the result display box.
+    1. Test case: `delete-o 4`
+       <br>Expected: No clorderient is deleted. Error details shown in the result display box.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1403]:
```suggestion
       <br>Expected: No order is deleted. Error details shown in the result display box.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+    1. Prerequisite (should be satisfied for each individual test case):
+       1. Order list does not contain an order with the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
+          * Order Deadline: 12 December 2021 at 12 pm
+          * Order Quantity: 1
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1332]: Add completion status

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+    1. Prerequisite (should be satisfied for each individual test case):
+       1. Order list contains at least one order and suppose the first order in the list has the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
+          * Order Deadline: 12 December 2021 at 12 pm
+          * Order Quantity: 1
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1437]: Add completion status

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+### Order
+1. Adding an Order
+    1. Prerequisite (should be satisfied for each individual test case):
+       1. Order list does not contain an order with the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1330]: Can you add `$` and `.00` for all the prices

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
     1. Test case: `list-c 123`
        <br>Expected: Client bookmarks list displays all clients.
 
+### Order
+1. Adding an Order
+    1. Prerequisite (should be satisfied for each individual test case):
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1323]:
```suggestion
    1. Prerequisites (should be satisfied for each individual test case):
```

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       <br>Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `delete-oi 1`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `delete-oi i/1`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+    1. Test case: `delete-oi`
+       <br>Expected: No order is edited. Error details shown in the result display box.
+
+1. Editing an order
+    1. Prerequisite (should be satisfied for each individual test case):
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1428]:
```suggestion
    1. Prerequisites (should be satisfied for each individual test case):
```

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Client Email: richard@gmail.com
+       1. Recipe bookmark list contains at least one recipe and suppose the first recipe in the list has the following details:
+          * Recipe Name: Nasi Lemak
+          * Recipe Ingredients: Rice x 200 g, Fish x 1 whole
+          * Recipe Price: 3
+    1. Test case: `edit-o 1 cn/Alexa White cp/91542652`
+       <br>Expected: First order in the order list is edited to have client name 'Alexa White' and client phone '91542652'.
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 c/1`
+       <br>Expected: First order in the order list is edited to have client details as the first client bookmark in the client bookmarks list.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1451]:
```suggestion
       <br>Expected: The details of the first client in the client bookmarks list is copied to the first order in the order list.
```

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Client Name: Richard Yeoh
+          * Client Phone: 96847225
+          * Client Address: Blk 2 Bedok Street 12. #06-41
+          * Client Email: richard@gmail.com
+       1. Recipe bookmark list contains at least one recipe and suppose the first recipe in the list has the following details:
+          * Recipe Name: Nasi Lemak
+          * Recipe Ingredients: Rice x 200 g, Fish x 1 whole
+          * Recipe Price: 3
+    1. Test case: `edit-o 1 cn/Alexa White cp/91542652`
+       <br>Expected: First order in the order list is edited to have client name 'Alexa White' and client phone '91542652'.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1448]:
```suggestion
       <br>Expected: The client name and phone in the first order is edited to 'Alexa White' and '91542652' respectively.
```

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 c/1`
+       <br>Expected: First order in the order list is edited to have client details as the first client bookmark in the client bookmarks list.
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 r/1`
+       <br>Expected: First order in the order list is edited to have recipe details as the first recipe bookmark in the recipe bookmarks list.
+       Details of the edited order are shown in the result display box.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1455]:
```suggestion
       <br>Expected: First order in the order list is edited to have recipe details as the first recipe bookmark in the recipe bookmarks list. Its position in the order list may change.
       Details of the edited order are shown in the result display box.
```

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Finding orders by keywords
+    1. Prerequisites: There are exactly 2 orders in the order list.
+       1. Suppose the first order has the following details:
+          * Client Name: Bernice Yu
+          * Client Phone: 98762159
+          * Client Address: Blk 12 Holland Village Street 1. #01-40
+          * Recipe Name: Nasi Lemak
+          * Recipe Ingredients: Rice x 200 g, Fish x 1 whole
+          * Order Price: 3
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1478]: Change price. See above

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Order Deadline: 15 December 2021 at 1 pm
+          * Order Quantity: 2
+          * Completion status: Uncompleted
+       1. Suppose the second order has the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1488]: Change price. See above

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Order Quantity: 1
+       1. Client bookmark list contains at least one client and suppose the first client in the list has the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Client Email: alexyeoh@gmail.com
+       1. Recipe bookmark list contains at least one recipe and suppose the first recipe in the list has the following details:
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Recipe Price: 4
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1341]: Change price. See above

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Editing an order
+    1. Prerequisite (should be satisfied for each individual test case):
+       1. Order list contains at least one order and suppose the first order in the list has the following details:
+          * Client Name: Alex Yeoh
+          * Client Phone: 87438807
+          * Client Address: Blk 30 Geylang Street 29. #06-40
+          * Recipe Name: Chicken Rice
+          * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+          * Order Price: 4
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1435]: Change price. See above

File: docs/DeveloperGuide.md [hunk 20]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+          * Order Quantity: 1
+       1. Client bookmark list contains at least one client and suppose the first client in the list has the following details:
+          * Client Name: Richard Yeoh
+          * Client Phone: 96847225
+          * Client Address: Blk 2 Bedok Street 12. #06-41
+          * Client Email: richard@gmail.com
+       1. Recipe bookmark list contains at least one recipe and suppose the first recipe in the list has the following details:
+          * Recipe Name: Nasi Lemak
+          * Recipe Ingredients: Rice x 200 g, Fish x 1 whole
+          * Recipe Price: 3
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1446]: Change price. See above

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -1318,6 +1318,195 @@ testers are expected to do more *exploratory* testing.
...
+       <br>Expected: First order in the order list is edited to have client name 'Alexa White' and client phone '91542652'.
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 c/1`
+       <br>Expected: First order in the order list is edited to have client details as the first client bookmark in the client bookmarks list.
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 r/1`
+       <br>Expected: First order in the order list is edited to have recipe details as the first recipe bookmark in the recipe bookmarks list.
+       Details of the edited order are shown in the result display box.
+    1. Test case: `edit-o 1 op/1 od/12-12-2021 1900 oq/2`
+       <br>Expected: First order in the order list is edited to have order price of '1', order deadline at 12 December 2021 7pm and order quantity of '2'.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1457]:
```suggestion
       <br>Expected: First order in the order list is edited to have order price of '1', order deadline at '12 December 2021 7pm' and order quantity of '2'.
```

**Review by Imranr2 — CHANGES_REQUESTED** [review]
Some nits to fix

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (sivayogasubramanian) [reply docs/DeveloperGuide.md:1386]: Its different

File: docs/DeveloperGuide.md (hunk 15 above)
Comment (sivayogasubramanian) [reply docs/DeveloperGuide.md:1455]: Why?

**Review by Imranr2 — APPROVED** [review]
LGTM!

**Pull request merged**